generators/csharp/base/src/asIs/QueryStringConverter.Template.cs

```diff
@@ -0,0 +1,230 @@
+        return queryCollection;
+    }
+
+    /// <summary>
+    /// Converts an object into a query string collection using Exploded Form notation.
+    /// </summary>
+    /// <param name="value">Object to form URL-encode. You can pass in an object or dictionary, but not lists, strings, or primitives.</param>
+    /// <exception cref="Exception">Throws when passing in a list, a string, or a primitive value.</exception>
+    /// <returns>A collection of key value pairs. The keys and values are not URL encoded.</returns>
+    internal static IEnumerable<KeyValuePair<string, string>> ToExplodedForm(object value)
+    {
+        var queryCollection = new List<KeyValuePair<string, string>>();
+        var json = JsonUtils.SerializeToElement(value);
+        AssertRootJson(json);
+        JsonToFormExploded(json, "", queryCollection);
+        return queryCollection;
+    }
+
+    /// <summary>
+    /// Converts an object into a query string collection using Form notation without exploding parameters.
+    /// </summary>
+    /// <param name="value">Object to form URL-encode. You can pass in an object or dictionary, but not lists, strings, or primitives.</param>
+    /// <exception cref="Exception">Throws when passing in a list, a string, or a primitive value.</exception>
+    /// <returns>A collection of key value pairs. The keys and values are not URL encoded.</returns>
+    internal static IEnumerable<KeyValuePair<string, string>> ToForm(object value)
+    {
+        var queryCollection = new List<KeyValuePair<string, string>>();
+        var json = JsonUtils.SerializeToElement(value);
+        AssertRootJson(json);
+        JsonToForm(json, "", queryCollection);
+        return queryCollection;
+    }
+
+    private static void AssertRootJson(JsonElement json)
+    {
+        switch (json.ValueKind)
+        {
+            case JsonValueKind.Object:
+                break;
+            case JsonValueKind.Array:
+            case JsonValueKind.Undefined:
+            case JsonValueKind.String:
+            case JsonValueKind.Number:
+            case JsonValueKind.True:
+            case JsonValueKind.False:
+            case JsonValueKind.Null:
+            default:
+                throw new Exception(
+                    $"Only objects can be converted to query string collections. Given type is {json.ValueKind}."
+                );
+        }
+    }
+
+    private static void JsonToForm(
+        JsonElement element,
+        string prefix,
+        List<KeyValuePair<string, string>> parameters
+    )
+    {
+        switch (element.ValueKind)
+        {
+            case JsonValueKind.Object:
+                foreach (var property in element.EnumerateObject())
+                {
+                    var newPrefix = string.IsNullOrEmpty(prefix)
+                        ? property.Name
+                        : $"{prefix}[{property.Name}]";
+
+                    JsonToForm(property.Value, newPrefix, parameters);
```

Comment:
Nice, this means we support deep object query params more broadly in C# now?

Reply:
I didn't include it in our raw client because we don't support it in our IR. Deep object was the first thing I implemented as I assumed that was just normal form encoding, but it is not.